tfx/experimental/pipeline_testing/base_stub_component_launcher.py

```diff
@@ -0,0 +1,119 @@
+from tfx.orchestration.launcher import in_process_component_launcher
+
+
+class BaseStubComponentLauncher(
+    in_process_component_launcher.InProcessComponentLauncher):
+  """Responsible for launching a stub executor.
+
+  A stub component launcher class inheriting BaseStubComponentLauncher can be
+  supported by the pipeline. This class should be defined in a separate python
+  module because the class is pickled by module path in beam and imported by
+  module path in KFP.
+  The executor will be launched in the same process of the rest of the
+  component, i.e. its driver and publisher.
+  """
+
+  stubbed_component_map = ...  # type: Dict[Text, Type[base_stub_executor.BaseStubExecutor]]
+  test_data_dir = ...  # type: Text
+
+  @classmethod
+  def initialize(
+      cls,
+      test_data_dir: Text,
+      stubbed_component_ids: List[Text],
+      stubbed_component_map: Dict[Text, Type[base_stub_executor.BaseStubExecutor]]  # pylint: disable=line-too-long
+  ):
+    """Intializes variables in the stub launcher class.
+
+    For beam pipeline, stub launcher class inheriting this class is defined in
+    tfx.experimental.pipeline_testing.stub_component_launcher.py
+
+    For KFP, stub launcher class inheriting this class is defined in
+    tfx.experimental.templates.taxi.launcher.stub_component_launcher.py.
+
+    These classes then can be used to launch stub executors in the pipeline.
+
+    For example,
+
+    class MyPusherStubExecutor(base_stub_executor.BaseStubExecutor){...}
+    class MyTransformStubExecutor(base_stub_executor.BaseStubExecutor){...}
+
+    stub_component_launcher.StubComponentLauncher.initialize(
+                test_data_dir,
+                stubbed_component_ids = ['CsvExampleGen'],
+                stubbed_component_map = {
+                    'Transform': MyTransformStubExecutor,
+                    'Pusher': MyPusherStubExecutor})
+    PipelineConfig(
+        supported_launcher_classes=[
+            stub_component_launcher.StubComponentLauncher
+        ],
+    )
+
+    The method initializes the necessary class variables for the
+    BaseStubComponentLauncher class, including stubbed_component_ids and
+    stubbed_component_map holding custom executor classes, which
+    users may define differently per component.
+
+    Args:
+      test_data_dir: The directory where pipeline outputs are recorded
+        (pipeline_recorder.py).
+      stubbed_component_ids: List of component ids that should be replaced
+        with aBaseStubExecutor.
+      stubbed_component_map: Dictionary holding user-defined stub executor.
+        These user-defined stub executors must inherit from
+        base_stub_executor.BaseStubExecutor.
+
+    Returns:
+      None
+    """
+    cls.stubbed_component_map = dict(stubbed_component_map)
+    for component_id in stubbed_component_ids:
+      cls.stubbed_component_map[component_id] = base_stub_executor.BaseStubExecutor  # pylint: disable=line-too-long
```

Comment:
I think that we can break this line into two lines instead of adding a pylint exception.